File: trunk/quote/Projects/PluginModel/ActiveHeader.vb
Imports System.ComponentModel

Imports Model.Common

Public Class ActiveHeader
    Implements INotifyPropertyChanged

    Private _Header As Header
    Public Shared ReadOnly ActiveHeader As ActiveHeader = New ActiveHeader

    Public Event PropertyChanged As PropertyChangedEventHandler _
        Implements INotifyPropertyChanged.PropertyChanged

    Public Property Header As Header
        Get
            Return _Header
        End Get
        Set(ByVal value As Header)
            If (_Header IsNot value) Then
                _Header = value
                RaiseEvent PropertyChanged(Me, _
                    New PropertyChangedEventArgs("Header"))
            End If
        End Set
    End Property

End Class

File: trunk/quote/Projects/PluginModel/ActiveHeader.cs
namespace Model
{
    using System;
    using System.ComponentModel;
    using System.Linq;
    
    using Model.Common;

    public class ActiveHeader : INotifyPropertyChanged
    {
        private static ActiveHeader instance = new ActiveHeader();
        
        private Header header;

        public event PropertyChangedEventHandler PropertyChanged;

        public static ActiveHeader Instance 
        {
            get { return instance; }
        }

        public Header Header 
        {
            get 
            { 
                return this.header; 
            }
            
            set 
            {
                if (!object.ReferenceEquals(this.header, value))
                {
                    this.header = value;
                    
                    if (this.PropertyChanged != null) 
                    {
                        this.PropertyChanged(this, new PropertyChangedEventArgs("Header"));
                    }
                }
            }
        }
    }
}